File: backend/ControleDeGastos/ControleDeGastos/Models/Transacao.cs
namespace ControleDeGastos.Models;

public class Transacao {
    public long Id { get; set; }
    public string Descricao { get; set; } = string.Empty;
    public decimal Valor { get; set; }
    public TipoTransacao Tipo { get; set; }
    public long PessoaId { get; set; }
    public Pessoa Pessoa { get; set; }
}

File: frontend/src/types/transacao.ts
export type TipoTransacao = "Despesa" | "Receita";

export interface Transacao {
  id: number;
  descricao: string;
  valor: number;
  tipo: TipoTransacao;
  pessoaId: number;
  pessoaNome: string;
}

export interface CriarTransacaoRequest {
  descricao: string;
  valor: number;
  tipo: TipoTransacao;
  pessoaId: number;
}
